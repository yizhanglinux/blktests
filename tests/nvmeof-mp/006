#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2016-2018 Western Digital Corporation or its affiliates

. tests/nvmeof-mp/rc

DESCRIPTION="Direct I/O with large transfer sizes and bs=8M"
QUICK=1

test_large_transfer_size() {
	local dev m

	use_blk_mq y || return $?
	dev=$(get_bdev 0) || return $?
	run_fio --verify=md5 --rw=randwrite --bs=8M --loops=$((10**6)) \
		--iodepth=4 --group_reporting --sync=1 --direct=1 \
		--ioengine=libaio \
		--filename="$dev" --name=large-io-test --thread --numjobs=1 \
		--runtime=10 --output="${RESULTS_DIR}/nvmeof-mp/fio-output-006.txt" \
		>>"$FULL"
}

test() {
	trap 'trap "" EXIT; teardown' EXIT
	setup && test_large_transfer_size && echo Passed
}
