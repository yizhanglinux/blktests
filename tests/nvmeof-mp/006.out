Configured NVMe target driver
Passed
